src/main/java/mekanism/common/util/EmitUtils.java

```diff
@@ -0,0 +1,73 @@
+package mekanism.common.util;
+
+import java.util.Set;
+import mekanism.common.base.SplitInfo;
+import mekanism.common.base.SplitInfo.DoubleSplitInfo;
+import mekanism.common.base.SplitInfo.IntegerSplitInfo;
+import mekanism.common.base.target.EnergyAcceptorTarget;
+import mekanism.common.base.target.Target;
+
+public class EmitUtils {
+
+    /**
+     * @param <HANDLER> The handler of our target.
+     * @param <TYPE> The type of the number
+     * @param <EXTRA> Any extra information we may need.
+     * @param <TARGET> The emitter target.
+     * @param availableTargets The targets to distribute toSend fairly among.
+     * @param totalTargets The total number of targets. Note: this number is bigger than availableTargets.size if any
+     * targets have more than one acceptor.
+     * @param splitInfo Information containing the split.
+     * @param toSend Any extra information such as gas stack or fluid stack.
+     * @return The amount that actually got sent.
+     */
+    public static <HANDLER, TYPE extends Number & Comparable<TYPE>, EXTRA, TARGET extends Target<HANDLER, TYPE, EXTRA>>
```

Comment:
This method could do with a brief explanation of its logic in the javadoc, since it uses variables that are changed outside this class

Reply:
Made it private for now as it is more of a helper generic method that doesn't need to be exposed anyways. And improved the comments inside it for describing how it works.